src/pipeline/config.cpp

```diff
@@ -253,7 +253,9 @@ namespace librealsense
                     return pdev->create_device();
             }
 
-            return ctx->add_device(file)->create_device();
+            auto dev_info = std::make_shared< playback_device_info >( ctx, file );
+            ctx->add_device( dev_info );
```

Comment:
Why not first creating the device and then adding it?

What will happen if we add it and then fail creating the device?

Reply:
I asked myself the same thing, but have not investigated -- that's how it was done before, so I didn't change anything

Reply:
I'll leave it to you to decide if a refactor is needed. :)